codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/endpoint/generators/EndpointParamsInterceptorGenerator.kt

```diff
@@ -156,6 +157,11 @@ class EndpointParamsInterceptorGenerator(
             when (node) {
                 is StringNode -> rust("Some(${node.value.dq()}.to_string())")
                 is BooleanNode -> rust("Some(${node.value})")
+                is ArrayNode -> {
+                    val elms = node.elements.map { "\"$it\".to_string()" }.joinToString(",")
```

Comment:
The two above use `node.value` instead. Is that load bearing? It feels odd that this one would just be `$it`

Reply:
I couldn't easily get the value because `$it` is just a generic `Node` type. I will figure out an idiomatic way to coerce it to `StringNode` so I can extract the value more explicitly.

Reply:
Added an (unsafe) cast in https://github.com/smithy-lang/smithy-rs/pull/3742/commits/d1cae4a89017a0bde9fb725329ef405329148ade which has the added benefit that this will fail if a non-string type is provided to the array. So we get a quick extra safety check out of it